tests/rules/no-dispatch-in-effects.test.ts

```diff
@@ -26,27 +45,155 @@ ruleTester().run(path.parse(__filename).name, rule, {
+      },
     ),
     fromFixture(
       stripIndent`
       import { Store } from '@ngrx/store'
       @Injectable()
       export class FixtureEffects {
-        effectNOK = createEffect(() => this.actions.pipe(
-          ofType('PING'),
-          map(() => this.customName.dispatch({ type: 'PONG' }))
-                    ~~~~~~~~~~~~~~~~~~~~~~~~ [${messageId}]
-          ), { dispatch: false })
+        effectNOK1 = createEffect(() => this.actions.pipe(
+          ofType(userActions.add),
+          map(() => {
+            return this.store.dispatch({ type: 'PONG' });
+                   ~~~~~~~~~~~~~~~~~~~ [${noDispatchInEffects}]
+          })
+        ))
 
-        constructor(private actions: Actions, private customName: Store){}
+        constructor(private actions: Actions, private store: Store) {}
       }`,
+      {
+        suggestions: [
+          {
+            messageId: noDispatchInEffectsSuggest,
+            output: stripIndent`
+            import { Store } from '@ngrx/store'
+            @Injectable()
+            export class FixtureEffects {
+              effectNOK1 = createEffect(() => this.actions.pipe(
+                ofType(userActions.add),
+                map(() => {
+                  return ({ type: 'PONG' });
+                })
+              ))
+
+              constructor(private actions: Actions, private store: Store) {}
+            }`,
+          },
+        ],
+      },
+    ),
+    fromFixture(
+      stripIndent`
+      import { Store } from '@ngrx/store'
+      @Injectable()
+      export class FixtureEffects {
+        effectNOK2 = createEffect(
+          () =>
+            this.actions.pipe(
+              ofType('PING'),
+              map(() => {
+                return this.customName.dispatch({ /* you shouldn't do this */ type: 'PONG' })
+                       ~~~~~~~~~~~~~~~~~~~~~~~~ [${noDispatchInEffects}]
+              }),
+            ),
+        )
+
+        constructor(private actions: Actions, private customName: Store) {}
+      }`,
+      {
+        suggestions: [
+          {
+            messageId: noDispatchInEffectsSuggest,
+            output: stripIndent`
+            import { Store } from '@ngrx/store'
+            @Injectable()
+            export class FixtureEffects {
+              effectNOK2 = createEffect(
+                () =>
+                  this.actions.pipe(
+                    ofType('PING'),
+                    map(() => {
+                      return ({ /* you shouldn't do this */ type: 'PONG' })
+                    }),
+                  ),
+              )
+
+              constructor(private actions: Actions, private customName: Store) {}
+            }`,
+          },
+        ],
+      },
+    ),
+    fromFixture(
+      stripIndent`
+      import { Store } from '@ngrx/store'
+      @Injectable()
+      export class FixtureEffects {
+        effectNOK3 = createEffect(
+          () =>
+            this.actions.pipe(
+              ofType('PING'),
```

Comment:
Btw, wouldn't it be good if we create a rule to avoid this kind of use?

Reply:
That's good for me 👍
If we do that, I think adding a similar one for `.dispatch()` would also make sense

Reply:
Actually `store.dispatch` doesn't accept a `string`, only `Action`.

Reply:
Correct, but it does support `store,dispatch({type: 'an action'})`.
And maybe we should also lint the old syntax (`store,dispatch(new Action())`) but this might be too harsh, so we could move  it behind a config?